clash-prelude/src/Clash/Signal/Delayed/Bundle.hs

```diff
@@ -219,3 +219,20 @@ instance KnownNat d => Bundle (RTree d a) where
   type Unbundled t delay (RTree d a) = RTree d (DSignal t delay a)
   bundle   = sequenceA
   unbundle = sequenceA . fmap lazyT
+
+-- | Internal data type, see "TaggedUnit".
+data TaggedUnit' (d :: k) (dom :: i) = TaggedUnit
+
+-- | Same as "Clash.Signal.Bundle", but adapted for "DSignal".
+type TaggedUnit (d :: Nat) (dom :: Domain) = TaggedUnit' '(d, d) '(dom, dom)
```

Comment:
Also, this being repeated here makes it essentially the same type in three places. At the very least I would expect the types for tags to be shared between normal / delayed `Bundle`, as the implementation should be identical.

Reply:
As discussed offline: both `TaggedUnit`s have a different number of type arguments (delayed vs non-delayed).